Feature/consolidation yggdrasil artifacts

V4L3 opened the pull request.

andreiciortea commented on lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/NotificationServerArtifact.java:
```diff
@@ -0,0 +1,234 @@
+                response.getWriter()
+                        .println("Link headers are missing! See the W3C WebSub Recommendation for details.");
+            } else {
+                /**
```
We should clean up this part

andreiciortea commented on lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/NotificationServerArtifact.java:
```diff
@@ -0,0 +1,234 @@
+import cartago.INTERNAL_OPERATION;
+import cartago.OPERATION;
+
+/**
```
We should document all operations exposed to agents

andreiciortea commented on lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/Notification.java:
```diff
@@ -0,0 +1,19 @@
+package org.hyperagents.jacamo.artifacts.yggdrasil;
+
+public class Notification {
```
We should document this class as well

andreiciortea commented on lib/src/main/java/org/hyperagents/jacamo/artifacts/yggdrasil/YggdrasilThingArtifact.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    /*
+     * Registers for WebSub to an Yggdrasil node. This is not a generic
```
Ideally, we should improve the implementation such that it is no longer specific to Yggdrasil — but works for any ThingArtifact using WebSub.

andreiciortea reviewed (changes requested):
Many thanks, @V4L3!

V4L3 commented:
@andreiciortea , thanks for the feedback.
I have made some adjustments:
- Added documentation
- Renamed YggdrasilThingArtifact to WebSubThingArtifact and moved it to the `wot` package
- Generalized the WebSub implementation to handle multiple cases: 2 Link headers, 1 Link header, Link tags in HTML
- Moved HMAS class to its own package
- Some Clean up
- Added an Example and adjusted the ReadMe.md

andreiciortea reviewed (approved):
Great, thanks @V4L3 !
